robottelo/hosts.py

```diff
@@ -0,0 +1,368 @@
+import logging
+from urllib.parse import urljoin
+from urllib.parse import urlunsplit
+
+from broker.hosts import Host
+
+from robottelo.config import settings
+from robottelo.constants import DISTRO_RHEL6
+from robottelo.constants import DISTRO_RHEL7
+from robottelo.constants import REPOS
+from robottelo.helpers import install_katello_ca
+from robottelo.helpers import remove_katello_ca
+
+logger = logging.getLogger(__name__)
+
+
+class ContentHostError(Exception):
+    pass
+
+
+class ContentHost(Host):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
```

Comment:
No need for a constructor here, `self.run` can be set as a class attribute to link it to execute.